x/launch/types/initial_genesis.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import "errors"
+
+// NewInitialGenesisDefault returns an initial genesis that is the default genesis
+func NewInitialGenesisDefault() *InitialGenesis {
+	var ig InitialGenesis
+
+	ig.Source = &InitialGenesis_DefaultGenesis{
+		DefaultGenesis: &DefaultInitialGenesis{},
+	}
+
+	return &ig
+}
+
+// NewInitialGenesisURL returns an initial genesis from a genesis URL
+func NewInitialGenesisURL(genesisURL GenesisURL) *InitialGenesis {
+	var ig InitialGenesis
+
+	ig.Source = &InitialGenesis_GenesisURL{
+		GenesisURL: &genesisURL,
+	}
+
+	return &ig
+}
+
+// GetType returns the type of the initial genesis
+func (ig InitialGenesis) GetType() (InitialGenesisType, error) {
```

Comment:
These enums seems to be defined in proto but never used in the source code or by grpc apis.